### DISCUSSION TITLE
Some simple code related issues in this config.

### DISCUSSION THREAD

**Discussion by charliie-dev** [ayamir/nvimdots#491] [Q&A] [answered]
1. `lua/keymap/bind.lua`
- What does `rhs_options` means?
- What does `pbind` means?
- Would their be some better naming?
- Would it be better to split up `keymap/init.lua` to multiple files as the [scopes](https://github.com/ayamir/nvimdots/wiki/Plugins#scopes) did? TBH, I think one mono-file is quite hard to read.
2. `lua/core/` -> `local vim = vim`?
- In various files in `lua/core/`, there's a `local vim = vim`, why do we need it? It seems fine to remove that line.

Thanks for answering!

**Comment by ayamir** [comment, marked as answer]
1. https://neovim.io/doc/user/map.html#mapping
2. `pbind` can be rename to `bind`.
3. Same as 2.
4. Sounds ok for more keymaps in future.
5. It's because we defined `globals = { "vim" }` for `sumneko_lua`. It will report warning if we remove this line.

**Reply by charliie-dev** [reply]
>     4. Sounds ok for more keymaps in future.

Will do a draft after.

>     5. It's because we defined `globals = { "vim" }` for `sumneko_lua`. It will report warning if we remove this line.

I'm  pretty sure it's the opposite way. We define `globals = { "vim" }` to avoid `sumneko` to report `vim` as invalid. At least, it's won't happen now.
I'll send a PR later.
